ksql-engine/src/test/java/io/confluent/ksql/KsqlEngineTest.java

```diff
@@ -535,6 +537,91 @@ public void shouldCleanUpInternalTopicSchemasFromSchemaRegistry() throws Excepti
     )));
   }
 
+  @Test
+  public void shouldCleanUpInternalTopicsOnClose() {
+    // Given:
+    final QueryMetadata query = ksqlEngine.execute(
+        "select * from test1;",
+        KSQL_CONFIG, Collections.emptyMap()).get(0);
+
+    query.start();
+
+    // When:
+    query.close();
```

Comment:
Shouldn't this be a test in QueryMetadataTest instead of here? It tests close() method for QueryMetadata.

Reply:
As @vcrfxia says, this is testing that the query the engine creates is correctly constructed so that when `close` is called on the query it calls back into the engine and the engine correctly cleans up internal topics.

Such a test does _not_ belong in the `QueryMetadataTest`. There is a test in there to test that `close` calls the supplied callback, that is all.

It would be wrong to switch this test to call `KsqlEngine::unregisterQuery` as the whole point is you don't call that any more. It's not a _private_ method automatically called when the query is closed.